examples/hpo/hpo_experiment.py

```diff
@@ -18,87 +19,75 @@ def hpo(config):
+        hpo = HPO_vizier(config.algo,
+                        config.task,
+                        config.load_study,
+                        config.output_dir,
+                        config.task_config,
+                        config.hpo_config,
+                        config.algo_config,
+                        config.safety_filter,
+                        config.sf_config,
+                        )
+    else:
+        raise ValueError('Only Optuna and Vizier are supported for now.')
+
 
     if config.hpo_config.hpo:
         hpo.hyperparameter_optimization()
         print('Hyperparameter optimization done.')
 
 
-def train(config):
-    '''Training for a given set of hyperparameters.
-
-    Usage:
-        * to start training, use with `--func train`.
-    '''
-    # Override algo_config with given yaml file
-    if config.opt_hps == '':
-        # if no opt_hps file is given
-        pass
-    else:
-        # if opt_hps file is given
-        with open(config.opt_hps, 'r') as f:
-            opt_hps = yaml.load(f, Loader=yaml.FullLoader)
-        for hp in opt_hps:
-            if isinstance(config.algo_config[hp], list) and not isinstance(opt_hps[hp], list):
-                config.algo_config[hp] = [opt_hps[hp]] * len(config.algo_config[hp])
-            else:
-                config.algo_config[hp] = opt_hps[hp]
-    # Experiment setup.
-    set_dir_from_config(config)
-    set_seed_from_config(config)
-    set_device_from_config(config)
-
-    # Define function to create task/env.
-    env_func = partial(make, config.task, output_dir=config.output_dir, **config.task_config)
-    # Create the controller/control_agent.
-    # Note:
-    # eval_env will take config.seed * 111 as its seed
-    # env will take config.seed as its seed
-    control_agent = make(config.algo,
-                         env_func,
-                         training=True,
-                         checkpoint_path=os.path.join(config.output_dir, 'model_latest.pt'),
-                         output_dir=config.output_dir,
-                         use_gpu=config.use_gpu,
-                         seed=config.seed,
-                         **config.algo_config)
-    control_agent.reset()
-
-    eval_env = env_func(seed=config.seed * 111)
-    # Run experiment
-    experiment = BaseExperiment(eval_env, control_agent)
-    experiment.launch_training()
-    results, metrics = experiment.run_evaluation(n_episodes=1, n_steps=None, done_on_max_steps=True)
-    control_agent.close()
-
-    return eval_env.X_GOAL, results, metrics
-
-
-MAIN_FUNCS = {'hpo': hpo, 'train': train}
+MAIN_FUNCS = {'hpo': hpo}
 
 
 if __name__ == '__main__':
     # Make config.
     fac = ConfigFactory()
-    fac.add_argument('--func', type=str, default='train', help='main function to run.')
-    fac.add_argument('--opt_hps', type=str, default='', help='yaml file as a result of HPO.')
+    fac.add_argument('--func', type=str, default='hpo', help='main function to run.')
```

Comment:
what other options are there other than 'hpo'?

Reply:
Deleted it.